schemas/core/customer.json

```diff
@@ -67,6 +67,40 @@
         }
       ]
     },
+    "balances": {
+      "type": "object",
+      "properties": {
+        "WMP": {
+          "type": "object",
+          "properties": {
+            "currency": { "const": "WMP" },
+            "amount": {
+              "type": "integer",
+              "minimum": 0
+            }
+          },
+          "required": ["currency", "amount"],
+          "additionalProperties": true
+        }
+      },
+      "patternProperties": {
+        "^[a-z]+(-[a-z]+)*-[a-z0-9_]+$": {
+          "type": "object",
+          "description": "key contains token id",
+          "properties": {
+            "currency": { "const": "TOKEN" },
+            "amount": {
+              "type": ["integer", "null"],
+              "minimum": 0
```

Comment:
There is also tokenId property
For example: `{ amount : 1, currency: TOKEN, tokenId: fi-whim-car-cdmr }`

Reply:
I know. However, `tokenId` is already present in the key. I decided to leave the redundant `tokenId` field out since JSON schema doesn't support checking redundant copies of the same information. It was going to be part of the spec but was left out https://stackoverflow.com/a/41192068

Reply:
We don't need to check for copies. It's enough having ambiguous `tokenId` property of type string

Reply:
Should we trust one or the other more if they are different for some reason? Should we document which one is authoritative? Does everyone need to verify separately that they are the same or is it ok to trust in your code that they are always the same? Is the redundant copy really necessary?

Reply:
The reason why user balances is an object is because it is easier to create database queries to modify its value. DB queries to modify an array is a pain to create and optimize

Because it is an object, there needs to be root key, and the only attribute that can be used is Currency/TokenId

Root key and tokenId are the same, and we can only trust that we ensure in the code that they stay the same.

Reply:
Ok. So technically the key should not be trusted since it is a token id "by accident".

Reply:
Well it is not "an accident". It is an old design decision that has no backing safeguard. If the key and tokenId are not the same, then there's a bug in the system.

Reply:
I added the field together with explanation

Reply:
awesome! :)